### ISSUE TITLE
Gradle 6.6, 6.6.1, 6.7-rc-1 - Gradle build daemon disappeared unexpectedly when using jsass 5.7.1 (works in gradle 6.5) Linux

### ISSUE THREAD

**Issue by fanick1** [gradle/gradle#14484]
tl;dr; I have prepared sample repository with easy repro steps to demonstrate the issue https://github.com/fanick1/gradle_libsass_issue

In essence using io.bit3:jsass:5.7.1 (or newer) inside  gradle 6.6  build fails with an error 
```
FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
```

### Expected Behavior
Successful build

### Current Behavior
Failed build, gradle build daemon crashes

### Context
Problem starts to appear with gradle 6.6 (and newer)
Problem does not appear with gradle 6.5 (and probably older)

For last two years we have been using jsass 5.7.3 in our builds to compile sass files into css files. We have integrated this library into a gradle plugin that adds a task that starts the sass compilation. With planned upgrade to gradle 6.6.1 in our build pipeline I have started to experience the issue described above on my Linux machine. It took some digging to figure out what causes the daemon to crash - libsass.so getting loaded by the jsass. Also it is worth noting that downgrading a bit back to jsass 5.7.0 seems like a good workaround for now.

### Steps to Reproduce 
`git clone https://github.com/fanick1/gradle_libsass_issue`
run `./gradlew explodes`

 there is also `./gradlew good` that runs against the older version of the libsass.so packed in the jsass 5.7.0 , running that task should not fail the build

### Your Environment
```
lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	Fedora
Description:	Fedora release 30 (Thirty)
Release:	30
Codename:	Thirty
```

```
java --version
java 11.0.4 2019-07-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.4+10-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.4+10-LTS, mixed mode)
```

```
ldd --version
ldd (GNU libc) 2.29
```

In the repo there is also sort of stack trace dump in stacktrace.txt.
```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007f53e689d895 in __GI_abort () at abort.c:79
#2  0x00007f53e68f608f in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7f53e6a04a7e "%s\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007f53e68fd40c in malloc_printerr (str=str@entry=0x7f53e6a02c22 "free(): invalid pointer") at malloc.c:5366
#4  0x00007f53e68fed5c in _int_free (av=<optimized out>, p=<optimized out>, have_lock=<optimized out>) at malloc.c:4185
#5  0x00007f52c3b121a2 in std::locale::_Impl::_M_install_facet(std::locale::id const*, std::locale::facet const*) () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#6  0x00007f52c3b10273 in std::locale::_Impl::_Impl(unsigned long) () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#7  0x00007f52c3b111e5 in std::locale::_S_initialize_once() () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#8  0x00007f53e6c67d7f in __pthread_once_slow (once_control=0x7f52c3dfbdb8 <std::locale::_S_once>, init_routine=0x7f52c3b111d0 <std::locale::_S_initialize_once()>) at pthread_once.c:116
#9  0x00007f52c3b11231 in std::locale::_S_initialize() () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#10 0x00007f52c3b11273 in std::locale::locale() () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#11 0x00007f52c3b124ec in std::ios_base::Init::Init() () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#12 0x00007f52c3954a2d in _GLOBAL__sub_I_ast.cpp () from /home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so
#13 0x00007f53e6cc6f4a in call_init (l=<optimized out>, argc=argc@entry=21, argv=argv@entry=0x7ffccbb3e758, env=env@entry=0x7ffccbb3e808) at dl-init.c:72
#14 0x00007f53e6cc7051 in call_init (env=0x7ffccbb3e808, argv=0x7ffccbb3e758, argc=21, l=<optimized out>) at dl-init.c:30
#15 _dl_init (main_map=main_map@entry=0x7f52cc039f10, argc=21, argv=0x7ffccbb3e758, env=0x7ffccbb3e808) at dl-init.c:119
#16 0x00007f53e6ccafae in dl_open_worker (a=a@entry=0x7f53903fdcd0) at dl-open.c:510
#17 0x00007f53e69b1dd9 in __GI__dl_catch_exception (exception=exception@entry=0x7f53903fdcb0, operate=operate@entry=0x7f53e6ccabc0 <dl_open_worker>, args=args@entry=0x7f53903fdcd0) at dl-error-skeleton.c:196
#18 0x00007f53e6cca82e in _dl_open (file=0x7f52cc03b0d0 "/home/x/gitrepos/libsass_issue/libjsass-5.7.1-libsass-3.5.3.so", mode=-2147483647, caller_dlopen=0x7f53e5fec821 <os::dll_load(char const*, char*, int)+65>, nsid=-2, argc=21, argv=<optimized out>, env=0x7ffccbb3e808) at dl-open.c:592
#19 0x00007f53e6a4239c in dlopen_doit (a=a@entry=0x7f53903fdef0) at dlopen.c:66
#20 0x00007f53e69b1dd9 in __GI__dl_catch_exception (exception=exception@entry=0x7f53903fde90, operate=operate@entry=0x7f53e6a42340 <dlopen_doit>, args=args@entry=0x7f53903fdef0) at dl-error-skeleton.c:196
#21 0x00007f53e69b1e73 in __GI__dl_catch_error (objname=objname@entry=0x7f52cc03b050, errstring=errstring@entry=0x7f52cc03b058, mallocedp=mallocedp@entry=0x7f52cc03b048, operate=operate@entry=0x7f53e6a42340 <dlopen_doit>, args=args@entry=0x7f53903fdef0) at dl-error-skeleton.c:215
#22 0x00007f53e6a42b09 in _dlerror_run (operate=operate@entry=0x7f53e6a42340 <dlopen_doit>, args=args@entry=0x7f53903fdef0) at dlerror.c:170
#23 0x00007f53e6a4242a in __dlopen (file=<optimized out>, mode=<optimized out>) at dlopen.c:87
#24 0x00007f53e5fec821 in os::dll_load(char const*, char*, int) () from /usr/java/jdk-11.0.4/lib/server/libjvm.so
#25 0x00007f53e5d1e0e1 in JVM_LoadLibrary () from /usr/java/jdk-11.0.4/lib/server/libjvm.so
#26 0x00007f53e4d6def8 in Java_java_lang_ClassLoader_00024NativeLibrary_load0 () from /usr/java/jdk-11.0.4/lib/libjava.so
#27 0x00007f53c8c2f890 in ?? ()
#28 0x00007f53c8c29680 in ?? ()
#29 0x0000000000000008 in ?? ()
#30 0x00000000eb1b9688 in ?? ()
#31 0x00007f53903fe9f8 in ?? ()
#32 0x0000000000000000 in ?? ()
```

This is way beyond my area of knowledge but after inspecting the stacktrace my first thought is that perhaps this is just a bug in the glibc ? I don't know...

**Comment by big-guy** [comment]
@wolfs @lptr does this look familiar to you?

**Comment by lptr** [comment]
I never seen an error similar to this, no. :/

**Comment by fanick1** [comment]
Just for the record I have filed an issue in the jsass project https://gitlab.com/jsass/jsass/-/issues/92 and also suggested a fix that seems to work https://gitlab.com/jsass/jsass/-/merge_requests/62, any 👍 votes on those two would be appreciated.

**Comment by big-guy** [comment]
I'm going to close this issue as it's not something we're likely to track down separately ourselves.

Maybe the [jsass gradle plugin](https://github.com/freefair/gradle-plugins/tree/master/jsass-plugin/src/main/java/io/freefair/gradle/plugins/jsass) could be changed to not load jsass into the Gradle daemon and instead fork a separate process (maybe with the worker API?).  That might make it less likely to conflict with changes on the Gradle side.
